### PR TITLE
StudentProfilePageUiTest failing on live server #8393

### LINKED ISSUE #8393: StudentProfilePageUiTest failing on live server
Error:
```
	at org.openqa.selenium.remote.ErrorHandler.createThrowable(ErrorHandler.java:206)
	at org.openqa.selenium.remote.ErrorHandler.throwIfResponseFailed(ErrorHandler.java:158)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:678)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:363)
	at org.openqa.selenium.remote.RemoteWebDriver.findElementByCssSelector(RemoteWebDriver.java:492)
	at org.openqa.selenium.By$ByCssSelector.findElement(By.java:430)
	at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:355)
	at org.openqa.selenium.support.pagefactory.DefaultElementLocator.findElement(DefaultElementLocator.java:69)
	at org.openqa.selenium.support.pagefactory.internal.LocatingElementHandler.invoke(LocatingElementHandler.java:38)
	at com.sun.proxy.$Proxy21.isDisplayed(Unknown Source)
	at org.openqa.selenium.support.ui.ExpectedConditions.elementIfVisible(ExpectedConditions.java:302)
	at org.openqa.selenium.support.ui.ExpectedConditions.access$100(ExpectedConditions.java:41)
	at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:288)
	at org.openqa.selenium.support.ui.ExpectedConditions$10.apply(ExpectedConditions.java:285)
	at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:238)
	at teammates.test.pageobjects.AppPage.waitForElementVisibility(AppPage.java:238)
	at teammates.test.pageobjects.GoogleLoginPage.submitCredentials(GoogleLoginPage.java:119)
	at teammates.test.pageobjects.GoogleLoginPage.completeGoogleLoginSteps(GoogleLoginPage.java:79)
	at teammates.test.pageobjects.GoogleLoginPage.loginAsInstructor(GoogleLoginPage.java:47)
	at teammates.test.cases.browsertests.BaseUiTestCase.loginInstructorToPage(BaseUiTestCase.java:144)
	at teammates.test.cases.browsertests.StudentProfilePageUiTest.getProfilePicturePage(StudentProfilePageUiTest.java:289)
	at teammates.test.cases.browsertests.StudentProfilePageUiTest.testAjaxPictureUrl(StudentProfilePageUiTest.java:273)
	at teammates.test.cases.browsertests.StudentProfilePageUiTest.allTests(StudentProfilePageUiTest.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:659)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:845)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1153)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:771)
	at org.testng.TestRunner.run(TestRunner.java:621)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:357)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:352)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:310)
	at org.testng.SuiteRunner.run(SuiteRunner.java:259)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1199)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1124)
	at org.testng.TestNG.run(TestNG.java:1032)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:72)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
```

### PR DESCRIPTION
**Pull request by tran-tien-dat**
Fixes #8393

**Outline of Solution**

As discussed in #8323, this PR fixes the `StudentProfilePageUiTest` class to use the instructor account with Google-Id correctly. Additionally, the test logic is also modified. To test the case when instructor does not have view-student-picture privilege (instructor is just a helper), a new course is created and the instructor lacks the privilege in this course.

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@whipermr5 can review since you're the one reviewing #8323?

File: src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java [hunk 1]
```diff
@@ -270,23 +289,31 @@ private void testAjaxPictureUrl() {
...
+
+        emailStudentForHelperCourse = StringHelper.encrypt(emailStudentForHelperCourse);
+        courseIdHelperCourse = StringHelper.encrypt(courseIdHelperCourse);
+        getProfilePicturePage(instructorGoogleId, TestProperties.TEST_INSTRUCTOR_PASSWORD,
+                emailStudentForHelperCourse, courseIdHelperCourse, NotAuthorizedPage.class);
 
         ______TS("Failure case: non-existent student");
 
-        getProfilePicturePage(instructorId, invalidEmail, courseId, EntityNotFoundPage.class);
+        getProfilePicturePage(instructorId, TestProperties.TEST_INSTRUCTOR_PASSWORD,
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java]: The test fails here on a staging server. There is no such Google account `SHomeUiT.instr`.

**Comment by tran-tien-dat** [comment]
My bad. It should be `instructorGoogleId`. I'll check the whole thing
again.

File: src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java [hunk 2]
```diff
@@ -270,23 +289,31 @@ private void testAjaxPictureUrl() {
...
         ______TS("Typical case: with email and course");
 
-        getProfilePicturePage(instructorId, email, courseId, StudentProfilePicturePage.class).verifyHasPicture();
+        getProfilePicturePage(instructorGoogleId, TestProperties.TEST_INSTRUCTOR_PASSWORD,
+                email, courseId, StudentProfilePicturePage.class).verifyHasPicture();
 
         ______TS("Failure case: instructor does not have privilege");
 
-        getProfilePicturePage(helperId, email, courseId, NotAuthorizedPage.class);
+        String emailStudentForHelperCourse = testData.students.get("studentForHelperCourse").email;
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java]: `studentForHelperCourseEmail` might read better.

File: src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java [hunk 3]
```diff
@@ -270,23 +289,31 @@ private void testAjaxPictureUrl() {
...
+        String courseIdHelperCourse = testData.students.get("studentForHelperCourse").course;
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/StudentProfilePageUiTest.java]: `helperCourseId` might read better.

**Comment by tran-tien-dat** [comment]
"helper" course here can be understood in 2 ways:
1. in a similar way we usually talk about helper classes: some minor classes to help us do something that is hard to expressed concisely. What exactly is that something is not specified and the information has to be obtained from somewhere else (e.g. comment in line 33)
2. the course where the instructor is a helper

The 2nd interpretation is hard to get from just the name. And similarly, the name "helped" course which implies the course in  which the instructor helped is also hard to get from just the name. So the reader has to look at other things to understand anyway. Hence, I don't think we should favour "helped" over "helper".

**Pull request merged**